js/phoenix/src/components/context/AuthContext.jsx

```diff
@@ -0,0 +1,56 @@
+import { createContext, useContext, useEffect, useState } from 'react'
+import {
+  onAuthStateChanged,
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  signOut,
+} from 'firebase/auth'
+import { auth } from '../../../config/firebase'
+
+const AuthContext = createContext({})
+
+export const useAuth = () => useContext(AuthContext)
+
+export const AuthContextProvider = ({
+  children,
+}) => {
+  const [user, setUser] = useState(null)
+  const [loading, setLoading] = useState(true)
+  //console.log(user)
+
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, (user) => {
+      if (user) {
+        setUser({
+          uid: user.uid,
+          email: user.email,
+          displayName: user.displayName,
+        })
+      } else {
+        setUser(null)
+      }
+      setLoading(false)
+    })
+
+    return () => unsubscribe()
+  }, [])
+
+  const signup = (email, password) => {
+    return createUserWithEmailAndPassword(auth, email, password)
+  }
+
+  const login = (email, password) => {
+    return signInWithEmailAndPassword(auth, email, password)
+  }
+
+  const logout = async () => {
+    setUser(null)
+    await signOut(auth)
+  }
+
+  return (
+    <AuthContext.Provider value={{ user, login, signup, logout }}>
+      {loading ? (<div>Loading ...</div>) : children}
```

Comment:
@StephenJakku Create a Loading component and replace this div

Reply:
Done